frontend/talentsearch/src/js/components/search/SearchForm.tsx

```diff
@@ -99,14 +97,22 @@ export const SearchForm: React.FunctionComponent<SearchFormProps> = ({
   );
   const assetMap = useMemo(() => mapIdToValue(cmoAssets), [cmoAssets]);
 
-  const formValuesToData = useCallback(
-    (values: FormValues): PoolCandidateFilterInput => {
+  // The location state holds the initial values plugged in from user. This is required if the user decides to click back and change any values.
+  const state = location.state as LocationState;
+  const initialValues = state ? state.some.initialValues : {};
+  const methods = useForm<FormValues>({ defaultValues: initialValues });
+  const { watch } = methods;
+
+  React.useEffect(() => {
+    const formValuesToData = (values: FormValues): PoolCandidateFilterInput => {
       return {
         classifications: values.classifications
-          ? values.classifications?.map((id) => classificationMap.get(id))
+          ? values.classifications?.map((id) =>
+              id ? classificationMap.get(id) : null,
+            )
```

Comment:
If there were a case where this returned null, I think it might break the api query. Maybe filter out nulls after? (Same for cmoAssets.)
```
          ? values.classifications
              ?.map((id) => (id ? classificationMap.get(id) : null))
              .filter(notEmpty)
```